experimental/list_formatter/src/patterns.rs

```diff
@@ -0,0 +1,129 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::list_formatter::{Type, Width};
+
+// This should be a compact representation of the CLDR data. Each locale entry is a 3 x 3 x 4 array 
+// (type x width x 4 patterns) of &str. These refs use 592 bytes of memory on a 64-bit platform,
+// plus the memory actually required by the strings. As there aren't many unique patterns, this is
+// probably negligible (see the test below).
+type RawPatterns<'a> = [[[&'a str;4];3];3];
```

Comment:
*Later:* This should eventually turn into a CLDR transformer.  Read about CLDR transformers and how to integrate this into the DataProvider pipeline here:

https://github.com/unicode-org/icu4x/blob/main/docs/tutorials/writing_a_new_data_struct.md#writing-a-new-data-struct

However, I would like to check in this PR as-is and then do the CLDR transformer in a follow-up.